pkg/controller.v1/mpi/mpijob_controller.go

```diff
@@ -1035,7 +1035,6 @@ func (jc *MPIJobReconciler) newLauncher(mpiJob *kubeflowv1.MPIJob, kubectlDelive
 		jc.PodGroupControl.DecoratePodTemplateSpec(podSpec, mpiJob, rt)
 	}
 
-	podSpec.Spec.ServiceAccountName = launcherName
```

Comment:
I just tried to build Training Operator image with your change and MPIJob test failed for me:
```
pytest sdk/python/test/e2e/test_e2e_mpijob.py --log-cli-level=info --namespace=default -k "test_sdk_e2e"
```

I think, we can't just remove SA assignment for MPIJob launcher.
MPIJob launcher [requires the appropriate RBAC to `exec`](https://github.com/kubeflow/training-operator/blob/4de96e7e6dbce9f1c7e947f3e74d3c8eabf94ae2/pkg/controller.v1/mpi/mpijob_controller.go#L1298-L1346) and access MPIJob worker pods.
Thus, we attach the created ServiceAccount to MPIJob launcher.

MPI Operator experts can comment on this @alculquicondor @tenzen-y @terrytangyuan

Reply:
It doesn't sound right to remove the SA.

That said, we got rid of it in v2, because we don't use `kubectl exec`